Assets/Scripts/Actor Components/Combat/Combat Abilities/MeleeAttackAbility.cs

```diff
@@ -17,8 +17,9 @@ public override void Execute(Combat combat, params object[] parameters)
         {
             combat.CombatStateMachine.ChangeState(new ReadyAttackState(combat, readyDuration, ReadyCallback));
         }
-        else
+        else if (combat.Resource.CanConsume(resourceCost))
         {
+            combat.Resource.Consume(resourceCost);
             combat.CombatStateMachine.ChangeState(new MeleeAttackState(combat, attackEffectArea));
```

Comment:
Some actors (enemies) are required to ready their melee attack before executing it. The actual execution occurs in the `ReadyCallback` method below this `Execute` method, so the resource check needs to occur there as well. Since there would be repeated logic in `Execute` and `ReadyCallback`, you can also consider consolidating these lines into a private method and call it from here and `ReadyCallback`.